packages/typespec-ts/test/integration/scalar.spec.ts

```diff
@@ -77,4 +77,122 @@ describe("Scalar Client", () => {
       assert.fail(err as string);
     }
   });
+
+  it("should get decimal type response body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal/response_body")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body, 0.33333);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put decimal type request body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal/resquest_body")
+        .put({ body: 0.33333 });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get decimal type request parameter", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal/request_parameter")
+        .get({ queryParameters: { value: 0.33333 } });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get decimal128 type response body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal128/response_body")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.strictEqual(result.body, 0.33333);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put decimal128 type request body", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal128/resquest_body")
+        .put({ body: 0.33333 });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get decimal128 type request parameter", async () => {
+    try {
+      const result = await client
+        .path("/type/scalar/decimal128/request_parameter")
+        .get({ queryParameters: { value: 0.33333 } });
```

Comment:
why 0.3333?

Reply:
This value is based on the [mockapi.ts](https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/type/scalar/mockapi.ts#L77).